build.gradle.kts

```diff
@@ -361,14 +361,19 @@ subprojects {
             }
         }
 
+        val dependencyManagement by configurations.creating {
+            isCanBeConsumed = false
+            isCanBeResolved = false
+            isVisible = false
+        }
+
         dependencies {
-            configurations.configureEach {
-                // Gradle and newer plugins will set these configuration properties correctly.
-                if (isCanBeResolved && !isCanBeConsumed
-                        // Older ones (like JMH) may not, so check the name as well.
-                        // Kotlin compiler classpaths don't support BOM nor need it.
-                        || name.endsWith("Classpath") && !name.startsWith("kotlin")) {
-                    add(name, platform(project(":dependencyManagement")))
+            add(dependencyManagement.name, platform(project(":dependencyManagement")))
+            afterEvaluate {
+                configurations.configureEach {
+                    if (isCanBeResolved && !isCanBeConsumed) {
+                        extendsFrom(dependencyManagement)
```

Comment:
This pattern from https://github.com/gradle/gradle/issues/12016#issuecomment-579177392

Somehow I landed on the previous code based on this comment, probably in an attempt to not add another configuration. But looking back at it, this version seems better anyways.

Reply:
Agreed.